taca/analysis/analysis.py

```diff
@@ -4,319 +4,209 @@
-                            CONFIG['analysis']['analysis_server']['sync']['data_archive']))
-                        transfer_run(run.run_dir)
-                    else:
-                        logger.info('Run {} already transferred to analysis server, skipping it'.format(run.id))
-                else:
-                    logger.warn('Run {} failed qc, transferring will not take place'.format(run.id))
-                    r_file = os.path.join(CONFIG['analysis']['status_dir'], 'report.out')
+                                            run.CONFIG['analysis_server']['host'],
+                                            run.CONFIG['analysis_server']['sync']['data_archive']))
+                run.transfer_run(CONFIG['storage']['archive_dirs'][run.sequencer_type], t_file,  False) #do not trigger analysis
+            else:
+                logger.warn('Run {} failed qc, transferring will not take place'.format(run.id))
+        elif run.is_transferred(t_file):
+            logger.info('Run {} already transferred to analysis server, skipping it'.format(run.id))
 
 
 
-        if not run.is_finished():
-            # Check status files and say i.e Run in second read, maybe something
-            # even more specific like cycle or something
-            logger.info('Run {} is not finished yet'.format(run.id))
-
     if run:
-        _process(Run(run))
+        #needs to guess what run type I have (HiSeq, MiSeq, HiSeqX)
+        sequencer_type = _run_type(run)
+        if sequencer_type is 'HiSeqX':
+            runObj = HiSeqX_Run(run, CONFIG["analysis"]["HiSeqX"])
+        elif sequencer_type is 'HiSeq':
+            runObj = HiSeq_Run(run, CONFIG["analysis"]["HiSeq"])
+        elif sequencer_type is 'MiSeq':
+            print "miseq: to be implemented soon"
+            runObj = None
+        else:
+            raise RuntimeError("New instrument type {}".format(sequencer_type))
+        _process(runObj, force_trasfer)
     else:
         data_dirs = CONFIG.get('analysis').get('data_dirs')
         for data_dir in data_dirs:
             runs = glob.glob(os.path.join(data_dir, '1*XX'))
+            # Try MiSeq runs as well
+            if not runs:
+                runs = glob.glob(os.path.join(data_dir, '1*000000000*'))
             for _run in runs:
-                _process(Run(_run))
+                sequencer_type = _run_type(_run)
+                if sequencer_type is 'HiSeqX':
+                    runObj = HiSeqX_Run(_run, CONFIG["analysis"]["HiSeqX"])
+                elif sequencer_type is 'HiSeq':
+                    runObj = HiSeq_Run(_run, CONFIG["analysis"]["HiSeq"])
+                elif sequencer_type is 'MiSeq':
+                    print "MiSeq: to be implemented soon"
+                    runObj = None
+                    #runObj = HiSeq_Run(_run, CONFIG["analysis"]["MiSeq"])
+                else:
+                    raise RuntimeError("New instrument type {}".format(sequencer_type))
```

Comment:
it can be done.. but. _process()  wants an Run type object that I need to initialise. My idea is to have _process as abstract as possible, it does the same thing on all Run types, so I do not like the idea that inside _process I need to guess the run type, I guess it outside and then I call it...